test/e2e/deployment.go

```diff
@@ -270,3 +274,131 @@ func testRollingUpdateDeploymentEvents(f *Framework) {
 	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up rc %s to 1", newRC.Name)))
 	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down rc %s to 0", rcName)))
 }
+
+func testRolloverDeployment(f *Framework) {
+	ns := f.Namespace.Name
+	c := f.Client
+	deploymentPodLabels := map[string]string{"name": "sample-pod"}
+	rcPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  "nginx",
+	}
+
+	rcName := "nginx-controller"
+	rcReplicas := 4
+	_, err := c.ReplicationControllers(ns).Create(&api.ReplicationController{
```

Comment:
This is almost same as ReplicationController in testRollingUpdateDeployment. Extract it out into a function rather than duplicating.